packages/rum-core/src/domain/action/trackClickActions.ts

```
@@ -141,9 +144,21 @@ function processPointerDown(
   pointerDownEvent: MouseEventOnElement,
   windowOpenObservable: Observable<void>
 ) {
-  const nodePrivacyLevel = configuration.enablePrivacyForActionName
-    ? getNodePrivacyLevel(pointerDownEvent.target, configuration.defaultPrivacyLevel)
-    : NodePrivacyLevel.ALLOW
+  const nodeSelfPrivacy = getNodePrivacyLevel(pointerDownEvent.target, configuration.defaultPrivacyLevel)
+  // If the node privacy level is MASK_UNLESS_ALLOWLISTED, we use the default privacy level
+  // If the enablePrivacyForActionName is true, we use the node privacy level
+  // Otherwise, we use the allow level
+  // TODO: we should make enablePrivacyForActionName true by default for the next major version
+
+  // this won't work because there would always be a case when we set the nodeSelfPrivacy to ALLOW
+  // if we use configuration.defaultPrivacyLevel === NodePrivacyLevel.MASK_UNLESS_ALLOWLISTED as a check
+  // then when we set defaultPrivacyLevel to ALLOW, and override the node privacy level to MASK_UNLESS_ALLOWLISTED
```

**thought:** I don't get this comment

> If the node privacy level is MASK_UNLESS_ALLOWLISTED, we use the default privacy level
> If the enablePrivacyForActionName is true, we use the node privacy level
> Otherwise, we use the allow level

That's exactly what's the code is doing. Could you explain why we do this instead?

> TODO: we should make enablePrivacyForActionName true by default for the next major version

Same here, why?

> this won't work because there would always be a case when we set the nodeSelfPrivacy to ALLOW
> if we use configuration.defaultPrivacyLevel === NodePrivacyLevel.MASK_UNLESS_ALLOWLISTED as a check
> then when we set defaultPrivacyLevel to ALLOW, and override the node privacy level to MASK_UNLESS_ALLOWLISTED

I guess something is missing here?

---

> TODO: we should make enablePrivacyForActionName true by default for the next major version

If we set this by default when the `configuration.defaultPrivacyLevel: MASK`, it would simplify things greatly and align with Session Replay.

The last paragraph was badly formed, I was explaining why we have to check for `isAllowlistMaskEnabled(configuration.defaultPrivacyLevel, nodeSelfPrivacy)`. When we set defaultPrivacyLevel to ALLOW, and override the node privacy level to MASK_UNLESS_ALLOWLISTED, without checking the nodeSelfPrivacy level we will not be able to apply the allowlist.